Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp

```diff
@@ -576,6 +576,12 @@ void AXIsolatedTree::updateNodeProperties(AXCoreObject& axObject, const Vector<A
         case AXPropertyName::SupportsSetSize:
             propertyMap.set(AXPropertyName::SupportsSetSize, axObject.supportsSetSize());
             break;
+        case AXPropertyName::TextInputMarkedRange:
+            if (auto textInputMarkedRange = axObject.textInputMarkedRange())
+                propertyMap.set(AXPropertyName::TextInputMarkedRange, *textInputMarkedRange);
+            else
+                propertyMap.remove(AXPropertyName::TextInputMarkedRange);
```

Comment:
It seems your intention here is if there is marked text, set the property, and otherwise remove it from the object's property map.

But unfortunately that won't work as you have. When you call `propertyMap.remove(AXPropertyName::TextInputMarkedRange);`, you're removing it from the local property map created in the beginning of this method, which is a no-op, and won't result in the property being removed from the object's map when `AXIsolatedTree::m_pendingPropertyChanges` is processed.

What we really need is a mechanism to remove a property from the map as a result of `AXIsolatedTree::updateNodeProperty`. But I don't want to block your change on that, so this should work in the meantime:

```
case AXPropertyName::TextInputMarkedRange: {
    auto textInputMarkedRange = axObject.textInputMarkedRange();
    // FIXME: We should have a mechanism to remove a property from the map for when textInputMarkedRange is std::nullopt.
    propertyMap.set(AXPropertyName::TextInputMarkedRange, textInputMarkedRange ? *textInputMarkedRange : nullptr);
    break;
}
```

Reply:
Oops, should have recognized that removing a key from a local here would have no effect when applying it to the original. Thanks for catching that, and I appreciate the suggestion. I'll apply it here.